R/engine.R

```diff
@@ -783,13 +783,34 @@ eng_comment = function(options) {}
 eng_verbatim = function(options) {
   # change default for the cat engine
   options$eval = FALSE
+  options = set_lang(options)
+  eng_cat(options)
+}
+
+set_lang = function(options, default = 'default') {
   # specify the lang name in engine.opts = list(lang = ), or lang/language,
   # or class.source; if all are empty, use 'default'
   options$engine.opts$lang = options$engine.opts$lang %n%
     unlist(options[c('lang', 'language')])[1] %n%
-    options$class.source %n% 'default'
+    options$class.source %n% default
+  options
+}
 
-  eng_cat(options)
+# embed a file verbatim
+eng_embed = function(options) {
+  # if `file` is empty, use `code` as the list of files
+  if (is.null(options$file)) {
```

Comment:
I think we need to assign the file option to `f` and just modify it if it is NULL
```suggestion
  if (is.null(f <- options$file)) {
```

Otherwise, this will error as f not define 
```
if (nchar(lang <- file_ext(f[1])) > 1) {
```